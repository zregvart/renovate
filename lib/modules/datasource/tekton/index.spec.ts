import { getPkgReleases } from '..';
import { Fixtures } from '../../../../test/fixtures';
import * as httpMock from '../../../../test/http-mock';
import { EXTERNAL_HOST_ERROR } from '../../../constants/error-messages';
import {
  defaultRegistryUrl,
  pipelineDatasource,
  taskDatasource,
} from './common';

const depName = 'buildpacks';

describe('modules/datasource/tekton/index', () => {
  describe('task', () => {
    describe('getReleases', () => {
      it('throws for 500', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Task`)
          .reply(500);
        await expect(
          getPkgReleases({
            datasource: taskDatasource,
            depName,
          })
        ).rejects.toThrow(EXTERNAL_HOST_ERROR);
      });

      it('returns null for empty 200 OK', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Task`)
          .reply(200, []);
        expect(
          await getPkgReleases({
            datasource: taskDatasource,
            depName,
          })
        ).toBeNull();
      });

      it('processes real data', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Task`)
          .reply(200, Fixtures.get('query-buildpack-task.json'));
        const res = await getPkgReleases({
          datasource: taskDatasource,
          depName,
        });
        expect(res).toMatchSnapshot();
        expect(res?.releases).toHaveLength(2);
      });
    });
  });

  describe('pipeline', () => {
    describe('getReleases', () => {
      it('throws for 500', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Pipeline`)
          .reply(500);
        await expect(
          getPkgReleases({
            datasource: pipelineDatasource,
            depName,
          })
        ).rejects.toThrow(EXTERNAL_HOST_ERROR);
      });

      it('returns null for empty 200 OK', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Pipeline`)
          .reply(200, []);
        expect(
          await getPkgReleases({
            datasource: pipelineDatasource,
            depName,
          })
        ).toBeNull();
      });

      it('processes real data', async () => {
        httpMock
          .scope(defaultRegistryUrl)
          .get(`/v1/query?name=${depName}&kind=Pipeline`)
          .reply(200, Fixtures.get('query-buildpack-pipeline.json'));
        const res = await getPkgReleases({
          datasource: pipelineDatasource,
          depName,
        });
        expect(res).toMatchSnapshot();
        expect(res?.releases).toHaveLength(1);
      });
    });
  });
});
