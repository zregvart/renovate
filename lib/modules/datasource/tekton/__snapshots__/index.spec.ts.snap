// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/datasource/tekton/index pipeline getReleases processes real data 1`] = `
Object {
  "homepage": "https://hub.tekton.dev",
  "registryUrl": "https://api.hub.tekton.dev/",
  "releases": Array [
    Object {
      "downloadUrl": "https://raw.githubusercontent.com/tektoncd/catalog/main/pipeline/buildpacks/0.2/buildpacks.yaml",
      "registryUrl": "https://api.hub.tekton.dev/",
      "releaseTimestamp": "2022-07-07T08:25:07.000Z",
      "version": "0.2",
    },
  ],
  "sourceUrl": "https://github.com/tektoncd/hub",
}
`;

exports[`modules/datasource/tekton/index task getReleases processes real data 1`] = `
Object {
  "homepage": "https://hub.tekton.dev",
  "registryUrl": "https://api.hub.tekton.dev/",
  "releases": Array [
    Object {
      "downloadUrl": "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks-phases/0.2/buildpacks-phases.yaml",
      "registryUrl": "https://api.hub.tekton.dev/",
      "releaseTimestamp": "2021-10-29T16:08:38.000Z",
      "version": "0.2",
    },
    Object {
      "downloadUrl": "https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks/0.5/buildpacks.yaml",
      "registryUrl": "https://api.hub.tekton.dev/",
      "releaseTimestamp": "2022-07-12T10:45:21.000Z",
      "version": "0.5",
    },
  ],
  "sourceUrl": "https://github.com/tektoncd/hub",
}
`;
